docs/tutorial_advanced.rst

```diff
@@ -680,13 +680,9 @@ The test will verify that all the nodes print the expected host name:
    :lines: 6-
    :emphasize-lines: 10-
 
-The first thing to notice in this test is that :attr:`~reframe.core.pipeline.RegressionTest.num_tasks` is set to zero.
-This is a requirement for flexible tests.
-The sanity check of this test simply counts the host names printed and verifies that they are as many as expected.
-Notice, however, that the sanity check does not use :attr:`num_tasks` directly, but rather access the attribute through the :func:`~reframe.utility.sanity.getattr` sanity function, which is a replacement for the :func:`getattr` builtin.
-The reason for that is that at the time the sanity check expression is created, :attr:`num_tasks` is ``0`` and it will only be set to its actual value during the run phase.
-Consequently, we need to defer the attribute retrieval, thus we use the :func:`~reframe.utility.sanity.getattr` sanity function instead of accessing it directly
-
+The first thing to notice in this test is that :attr:`~reframe.core.pipeline.RegressionTest.num_tasks` is set to zero as default, which is a requirement for flexible tests.
+However, this value is set to the actual number of tasks during the ``run`` pipeline stage.
```

Comment:
I would move this after the following sentence (starting without the "lastly") and rephrase it to something that the number of tasks is set after the job finishes to the actual number of tasks that were used.

Reply:
I'm not sure that breaking the chronological order of things would make this more clear. I've reworded this a bit.